lib/puppet/application.rb

```diff
@@ -209,7 +209,14 @@ def option_parser_commands
     # @return [Array<String>] the names of available applications
     # @api public
     def available_application_names
-      @loader.files_to_load.map do |fn|
+      # Use our configured environment to load the application, as it may
+      # be in a module we installed locally, otherwise fallback to our
+      # current environment (*root*). Once we load the application the
+      # current environment will change from *root* to the application
+      # specific environment.
+      environment = Puppet.lookup(:environments).get(Puppet[:environment]) ||
```

Comment:
Picky Pet Peeve - use of "our" and "we" in comments - who is that? An instance of a class, puppet's etc. Here "the configured environment" would read better. And say "in a module installed locally" (does not have to be "we" that installed, and then "fallback to the current environment".